scripts/entity/components/movement_components/player_movement_component.gd

```diff
@@ -63,12 +80,25 @@ func on_jump_input() -> void:
 
 func on_jump_input_cancelled() -> void:
 	_is_jump_button_held = false
+	
+func _on_entered_climbable() -> void:
+	_is_climbing = true
+	_can_climb = true
+	print("Entered climbable area")
+
+func _on_exited_climbable() -> void:
+	_is_climbing = false
+	_can_climb = false
+	if _is_up_pressed:
+		velocity.y = -LAUNCH_VELOCITY
```

Comment:
Personal opinion is you should have to put an input in to "jump" if exiting a ladder. I dont want to get to the top of a ladder and jump if I just want to walk right.

Reply:
Changed to an exported variable, can be set to 0 if an immediate stop is what the designers want